serenata_toolbox/presences.py

```diff
@@ -1,143 +1,165 @@
-        condition = extract_text(deputy, 'condicao')
-        if condition == 'Titular':
-            condition = 'Holder'
-        else:
-            condition = 'Substitute'
-        yield [
-            extract_text(deputy, 'ideCadastro'),
-            budget_id,
-            extract_text(deputy, 'matricula'),
-            condition,
-            extract_text(deputy, 'nome'),
-            extract_text(deputy, 'nomeParlamentar'),
-            extract_text(deputy, 'uf'),
-            extract_text(deputy, 'partido'),
-        ]
-
-records = parse_deputies(t.getroot())
-deputies = pd.DataFrame(records, columns=[
-    'congressperson_id',
-    'budget_id',
-    'registration',
-    'condition',
-    'civil_name',
-    'name',
-    'state',
-    'party'
-])
-print("Total deputies:", deputies.shape[0])
-print("Holder deputies:", deputies[deputies['condition'] == 'Holder'].shape[0])
-print("Substitute deputies:", deputies[deputies['condition'] == 'Substitute'].shape[0])
-
-###############################################################################
-## Presence in sessions
-
-url = (
-    'http://www.camara.leg.br/SitCamaraWS/sessoesreunioes.asmx/ListarPresencasParlamentar'
-    '?dataIni=01/02/2015'
-    '&dataFim=31/12/2016'
-    '&numMatriculaParlamentar={0}'
-)
-
-def fetch_presences(filtered_deputies):
-    print("Fetching data for", filtered_deputies.shape[0], "deputies")
-    for i, deputy in filtered_deputies.iterrows():
-        print(i, deputy['name'])
-        file = urllib.request.urlopen(url.format(deputy['registration']))
-        t = ET.ElementTree(file=file)
-        congressperson_name = extract_text(t.getroot(), 'nomeParlamentar')
-        for day in t.getroot().findall('.//dia'):
-            date = extract_datetime(day, 'data')
+import pandas as pd
+
+from serenata_toolbox import datasets
+from serenata_toolbox.cleanup import xml_extract_text, \
+                                     xml_extract_datetime, \
+                                     xml_extract_date, \
+                                     translate_column
+
+class Presences:
+
+    URL = (
+        'http://www.camara.leg.br/SitCamaraWS/sessoesreunioes.asmx/ListarPresencasParlamentar'
+        '?dataIni={}'
+        '&dataFim={}'
+        '&numMatriculaParlamentar={}'
+    )
+
+    def fetch(self, deputies, start_date, end_date):
+        """
+        :param deputies: (pandas.DataFrame) a dataframe with deputies data
+        :param date_start: (str) date in the format dd/mm/yyyy
+        :param date_end: (str) date in the format dd/mm/yyyy
+        """
+        print("Fetching data for {} deputies from {} -> {}".format(len(deputies), start_date, end_date))
+
+        records = self.__all_presences(deputies, start_date, end_date)
+
+        df = pd.DataFrame(records, columns=(
+            'term',
+            'congressperson_document',
+            # TODO: Rename to something else to avoid confusion (maybe just congressperson?)
```

Comment:
Yeah, `congressperson_name` looks like `Luana Costa-PSB/MA`, but we already have separate columns for state and party so I guess the best we can do is either rename it to something else to avoid confusion while keeping the dataset as close to the raw source as possible; or we can remove the party and state from the field and keep just the name around.

@Irio @cuducos Thoughts?